external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -319,8 +320,10 @@ private boolean emitTupleIfNotEmitted(ConsumerRecord<K, V> record) {
                 } else {
                     collector.emit(tuple, msgId);
                 }
-                emitted.add(msgId);
-                numUncommittedOffsets++;
+                if(!consumerAutoCommitMode){//only need to track in none-AutoCommitMode
+                    emitted.add(msgId);
```

Comment:
I think we still want to track emitted even in auto commit mode.  Replaying a failed tuple will not work unless it is added to emitted and even with auto commit some people may want to replay tuples.

numUncommittedOffsets is only ever decremented in auto commit mode so that part of the change is fine.  Although, it would be good to add a javadoc to numUncommittedOffsets indicating that it is not modified in auto commit mode.